hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java

```diff
@@ -82,7 +84,12 @@ public JobMetrics getMetrics() {
         try {
             ClientMessage response = invocation(request, masterAddress()).invoke().get();
             JetGetJobMetricsCodec.ResponseParameters parameters = JetGetJobMetricsCodec.decodeResponse(response);
-            return serializationService().toObject(parameters.response);
+            List<RawJobMetrics> shards = serializationService().toObject(parameters.response);
+            JobMetrics metrics = JobMetrics.empty();
+            for (RawJobMetrics shard : shards) {
+                metrics = metrics.merge(JobMetrics.of(shard.getTimestamp(), shard.getValues()));
+            }
```

Comment:
We recreate the map after each member using a materialized sub-map. Would be better to build the internal map directly using the iterator.